website/docs/plugin/sdkv2/logging/http-transport.mdx

```diff
@@ -0,0 +1,211 @@
+
+The recommended way to produce logs from your provider is using **[terraform-plugin-log](https://www.terraform.io/plugin/log)** library, and that provides [log filtering](https://www.terraform.io/plugin/log/filtering) functionality. So, we built a new _RoundTripper_ to deprecate the existing one.
+
+
+# Setting up logging for HTTP Transactions
+
+SDK offers helpers to put in place logging of HTTP Transactions: it’s built on top of **[terraform-plugin-log](https://www.terraform.io/plugin/log)**, so it allows to leverage its features, without having to write the whole implementation of `http.RoundTripper`.
+
+There are 2 functions inside the package `helper/logging`, each targeting a specific logging setup that the provider developer might have chosen (see [“Writing Log Output”](https://www.terraform.io/plugin/log/writing) for details):
+
+* `NewLoggingHTTPTransport(transport http.RoundTripper)`
+* `NewSubsystemLoggingHTTPTransport(subsystem string, transport http.RoundTripper)`
+
+Using one or the other depends on the developer, and if they want their logging to happen against the `tflog` Provider root logger, or against a `tflog` Provider [Subsystem logger](https://www.terraform.io/plugin/log/writing#subsystems). For the latter, the only thing to remember is that the `subsystem` string used with `NewSubsystemLoggingHTTPTransport()`, has to match the [pre-created subsystem logger name](https://www.terraform.io/plugin/log/writing#create-subsystems).
+
+To make use of our logging transport, there are 2 things that a provider developer should do:
+
+* Setting up the HTTP Client to use this new Transport
+* Ensuring that each HTTP Request, before being submitted to the Client, carries the configured `context.Context`, storing the desired logging configuration
+
+
+## Creating HTTP Client that uses the new logging transport
+
+Once the correct _Transport _has been created, it should be used when setting up the `http.Client` that the provider is going to use. For example, a good place to set up the client could be the `schema.Provider` `ConfigureContextFunc`:
+
+```go
+func New() (*schema.Provider, error) {
+    return &schema.Provider{
+        // omitting the rest of the schema definition
+
+        ConfigureContextFunc: func (ctx context.Context, rsc *schema.ResourceData) (interface{}, diag.Diagnostics) {
+
+            // omitting provider-specific configuration logic
+
+            transport := logging.NewLoggingHTTPTransport(http.DefaultTransport)
+            client := http.Client{
+                Transport: transport,
+            }
+
+            return client, diag.Diagnostics{}
+        }
+    }
+}
+```
+
+This will set up a client that is identical to the default Golang `http.Client`, except it uses the new logging transport.
```

Comment:
We want to explain what's going to happen in an example right before. That way, the user is primed to understand what we're about to show them!

```suggestion
After you create the transport , you must use to set up the `http.Client` for the provider. The following example sets up the client in `schema.Provider` `ConfigureContextFunc`. The client is identical to the default Golang `http.Client`, except it uses the new logging transport.

```go
func New() (*schema.Provider, error) {
    return &schema.Provider{
        // omitting the rest of the schema definition

        ConfigureContextFunc: func (ctx context.Context, rsc *schema.ResourceData) (interface{}, diag.Diagnostics) {

            // omitting provider-specific configuration logic

            transport := logging.NewLoggingHTTPTransport(http.DefaultTransport)
            client := http.Client{
                Transport: transport,
            }

            return client, diag.Diagnostics{}
        }
    }
}
```
```